docs/sources/next/javascript-api/k6-experimental/fs/file/read.md

```diff
@@ -0,0 +1,109 @@
+---
+title: 'read'
+excerpt: 'the read method is used to read a chunk of the file.'
+weight: 20
+---
+
+# read
+
+The `read` method is used to read a chunk of the file into an [Uint8Array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Uint8Array) buffer.
+
+It resolves to either the number of bytes read during the operation, or `null` if there was nothing more to read.
+
+## Parameters
+
+| Parameter | Type                                                                                                      | Description                       |
+| :-------- | :-------------------------------------------------------------------------------------------------------- | :-------------------------------- |
+| buffer    | [Uint8Array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Uint8Array) | The buffer to read the data into. |
+
+## Returns
+
+A [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) resolving to the number of bytes read or `null` if the end of the file has been reached.
+
+## Example
+
+### Reading a file
+
+In the following example, we open a file and read it in chunks of 128 bytes until we reach the end of the file.
+
+{{< code >}}
```

Comment:
Just as a comment, the `{{< code >}}` shortcode main purpose is to support multiple code snippets (how to do X in JavaScript, and in Ruby, for example). It still works if you have a single code snippet inside of it, but we don't necessarily need it here. 😄

Reply:
Oh, that's good to know! I think it still makes sense to have it even with a single example? Makes it more explicit too as it will display "Javascript" above the code right?